src/main/java/org/elasticsearch/transport/Transport.java

```diff
@@ -43,6 +44,12 @@
     BoundTransportAddress boundAddress();
 
     /**
+     * Further profile bound addresses
+     * @return Should return null if transport does not support profiles, otherwise a map with name of profile and its bound transport address
+     */
+    Map<String, BoundTransportAddress> profileBoundAddresses();
```

Comment:
does this make this issue a breaking one? I know, always the same dicussion but I never remember the outcome, it potentially breaks plugins that implement their own transport...